tests/testthat/test-Model-validity.R

```diff
@@ -0,0 +1,46 @@
+# validate_model_general ----
+
+test_that("validate_model_general passes for valid object", {
+  object <- .GeneralModel(
+    datanames = "x",
+    datamodel = function(x) {
+      x
+    },
+    priormodel = function(x) {
+      x
+    },
+    modelspecs = function(x) {
+      x
+    },
+    init = function(x) {
+      x
+    },
+    sample = "param1"
+  )
```

Comment:
maybe put this in another test helper function that gives you a `GeneralModel` object back?

Reply:
Sure, but I think these 2 places would be only the one we need it, and e.g. for .GeneralData we did not create such helper. Let me know if you still prefer do make it :)